lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -231,6 +227,22 @@ def store(payment_method, options = {})
         end
       end
 
+      def customer(post, payment, options)
+        post[:description] = options[:description] if options[:description]
+        post[:expand] = [:sources]
+
+        if billing = options[:billing_address] || options[:address]
+          post.merge!(add_address(post, billing, options))
+        end
+
+        if shipping = options[:shipping_address]
+          post[:shipping] = add_address({}, shipping, options)
+        end
```

Comment:
Why do we sometimes merge the result of `add_address` and sometimes it not?

Reply:
For customer billing address needs to be in the [address hash ](https://stripe.com/docs/api/customers/object#customer_object-address) and for shipping in needs to be in the [shipping hash](https://stripe.com/docs/api/customers/object#customer_object-shipping)